schemas/session-replay/segment-schema.json

```diff
@@ -0,0 +1,115 @@
+                    "description": "Application properties",
+                    "required": [
+                        "id"
+                    ],
+                    "properties": {
+                        "id": {
+                            "type": "string",
+                            "description": "UUID of the application",
+                            "pattern": "^[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$",
+                            "readOnly": true
+                        }
+                    },
+                    "readOnly": true
+                },
+                "session": {
+                    "type": "object",
+                    "description": "Session properties",
+                    "required": [
+                        "id"
+                    ],
+                    "properties": {
+                        "id": {
+                            "type": "string",
+                            "description": "UUID of the session",
+                            "pattern": "^[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$",
+                            "readOnly": true
+                        }
+                    },
+                    "readOnly": true
+                },
+                "view": {
+                    "type": "object",
+                    "description": "View properties",
+                    "required": [
+                        "id"
+                    ],
+                    "properties": {
+                        "id": {
+                            "type": "string",
+                            "description": "UUID of the view",
+                            "pattern": "^[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$",
+                            "readOnly": true
+                        }
+                    },
+                    "readOnly": true
+                },
+                "start": {
+                    "type": "integer",
+                    "description": "The start UTC timestamp in milliseconds corresponding to the first record in the Segment data. Each timestamp is computed as the UTC interval since 00:00:00.000 01.01.1970.",
+                    "readOnly": true
+                },
+                "end": {
+                    "type": "integer",
+                    "description": "The end UTC timestamp in milliseconds corresponding to the last record in the Segment data. Each timestamp is computed as the UTC interval since 00:00:00.000 01.01.1970.",
+                    "readOnly": true
+                },
+                "has_full_snapshot_record": {
+                    "type": "boolean",
+                    "description": "Whether this Segment contains a fullsnapshot record or not.",
+                    "readOnly": true
+                },
+                "records_count": {
+                    "type": "integer",
+                    "description": "The number of records in this Segment.",
```

Comment:
maybe a dumb question: why do we need `records_count` field when we have `records` array field and can simply know the count? Is it because records for the same view may arrive in different requests?

Reply:
Good question, is part of the `Segment` web event contract. Maybe @ThibautGeriz, @acorretti can give us more info on this matter.

Reply:
The current schema doesn't make this clear, but these properties are part of what's called the `SegmentMetadata` that is not only present in each segment, but also as individual attachments in the _multipart POST_ request to the intake endpoint.

It's like this so the intake can still know some details about the segment without having to decompress them.

Maybe it's a good idea to define the _segment metadata_ separately, then have _segment-schema_ extend it?

Reply:
> Good question, is part of the Segment web event contract. can give us more info on this matter.

As Alfonso said, they is the segment that's compressed and is described here but we also send the metadata in the HTTP payload as multi part field in order to index them in the event platform without having to decompress the payload stored on S3. So the `records` field in only in S3 while `records_count` is also in the event platform
 
> Maybe it's a good idea to define the segment metadata separately, then have segment-schema extend it?

I don't think it's worth